Support Fleet API and Proxy

andig opened the pull request:
Fix https://github.com/bogosj/tesla/issues/125

TODO

- [ ] fix examples

andig committed:
Drop vehicle ID

andig committed:
Remove streaming endpoint

andig committed:
Update go.mod

andig committed:
Add token source and client options

andig committed:
Drop custom transport

andig committed:
Drop auth handling and login command

andig committed:
Drop user agent

andig committed:
Drop oauth2 config and token options

andig commented:
We‘re testing this against the proxy now.

andig committed:
Allow updating base url after client has been created

andig committed:
Add user region api

andig committed:
Fix region

andig committed:
Remove outdated option

andig committed:
Add user me

andig committed:
Decode command reason when not HTTP 200

andig committed:
Better command error handling

andig commented on client.go:
```diff
@@ -104,34 +85,35 @@ func (c Client) getJSON(url string, out interface{}) error {
 	defer res.Body.Close()
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return body, err
```
```suggestion
		return nil, err
```

andig committed:
Update client.go

andig committed:
Update tests

andig committed:
Fix vehicle asleep handling

andig commented:
@bogosj our first tests are positive now. Except for the examples this would be a good time to review the changes and make a decision if you want to continue hosting the module.
In terms of api I'm not super happy with the `Set*Url` methods but need them to switch between fleet api directly and proxy. In addition it would also be nice to swap the client for an existing vehicle (in case vehicle was retrieved reading from fleet api but one would like to send commands to using proxy).

bogosj commented:
I have no objection to continuing to keep the library as bogosj/tesla. I'm probably not going to go to the effort to update my only use of this library, so I have no vested interest in keeping it alive. Perhaps it would be best with this major breaking change to take the opportunity to re-home it?

andig committed:
Allow changing vehicle client

andig committed:
Merge branch 'bogosj:main' into main

andig committed:
Return request creation errors instead of panicking (#3)

http.NewRequest returns a nil request when the url cannot be parsed. Both
get and post discarded that error and passed the nil request on, so
setHeaders dereferenced it and the process crashed. A mistyped proxy base
url was enough to take down the calling application.

andig committed:
Remove examples using deleted token file auth (#4)
